corehq/apps/users/views/__init__.py

```diff
@@ -1437,11 +1437,11 @@ def format_changes(domain, changed_values):
         program_id = changed_values.pop('program_id', None)
         if program_id:
             changed_values.update(UserChangeMessage.program_change(Program.get(program_id)))
-        assigned_locations = changed_values.pop('assigned_locations', None)
-        if assigned_locations:
+        if 'assigned_locations' in changed_values:
```

Comment:
Earlier we were ignoring if assigned_locations was passed as an empty list, which would be in case the assigned locations are wiped out. So, the change here is to accept whatever was passed, if it was passed.

Reply:
@esoergel just wanted to draw your attention to this specifically. `InvitationHistory` isn't exposed on the UI like UserHistory so this probably went unnoticed.

Reply:
This PR was originally intended to fix the incorrect updates to location information on web user invites.
I am thinking if the fixes for Invitation History should be done separately or this is okay to be included here.
Open to suggestions.

Reply:
I think that's probably fine to include here

Reply:
> There are two other places where this method is called out from, other than the changes being made here.
One of them is an API end point for invitations and it appears to be having the same bug that this PR is trying to resolve with bulk upload

The API end point does not allow update but only create so this bug does not happen here. The bug was when locations were removed intentionally in an update. For a creation of an invite, if locations are assigned or not assigned, both will work okay.